apps/src/flowlab/views/flow/nodes/WebNode.tsx

```diff
@@ -0,0 +1,81 @@
+import {
+  Handle,
+  Position,
+  useNodeConnections,
+  useNodesData,
+  type NodeProps,
+  useReactFlow,
+  NodeResizer,
+} from '@xyflow/react';
+import React, {memo, useEffect} from 'react';
+
+import {isTextNode, type MyNode} from '../../../flow/initialElements';
+
+// This node renders HTML sent to its input node as text in an iframe.
+// It is able to send a message received from its iframe to its output handle.
+
+function WebNode({id, selected}: NodeProps) {
+  const connections = useNodeConnections({
+    handleType: 'target',
+  });
+  const nodesData = useNodesData<MyNode>(
+    connections.map(connection => connection.source)
+  );
+  const textNodes = nodesData.filter(isTextNode);
+  const {updateNodeData} = useReactFlow();
+
+  useEffect(() => {
+    window.addEventListener('message', function (event) {
```

Comment:
How does this work exactly? Where is the `message` being sent from? Also does this mean a new event handler is being attached for each `WebNode`? Wouldn't they all respond to all messages in that case?